### PR TITLE
SONARJAVA-656 S1451/"Copyright and license headers" should accept a regex license

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java [hunk 1]
```diff
@@ -48,6 +50,7 @@
 public class FileHeaderCheck extends SubscriptionBaseVisitor implements CharsetAwareVisitor {
 
   private static final String DEFAULT_HEADER_FORMAT = "";
+  private final String MESSAGE = "Add or update the header of this file.";
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java]: MAJOR Rename this field name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 major

File: java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java [hunk 2]
```diff
@@ -72,20 +82,44 @@ public void setCharset(Charset charset) {
   @Override
   public void scanFile(JavaFileScannerContext context) {
     super.context = context;
-    expectedLines = headerFormat.split("(?:\r)?\n|\r");
+    if (isRegularExpression) {
+      try {
+        searchPattern = Pattern.compile(headerFormat, Pattern.DOTALL);
+      } catch (RuntimeException e) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java]: seems quite wide to catch all RuntimeException.

File: java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java [hunk 3]
```diff
@@ -72,20 +82,44 @@ public void setCharset(Charset charset) {
   @Override
   public void scanFile(JavaFileScannerContext context) {
     super.context = context;
-    expectedLines = headerFormat.split("(?:\r)?\n|\r");
+    if (isRegularExpression) {
+      try {
+        searchPattern = Pattern.compile(headerFormat, Pattern.DOTALL);
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/FileHeaderCheck.java]: There should be no need to recompile the RegExp for every file.

**Comment by mpaladin** [comment]
@benzonico recommended changes done.

**Comment by benzonico** [comment]
Except for a minor comment, LGTM.

**Pull request merged**